packages/plugins/plugin-wnfs/src/components/FileInput.tsx

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright 2025 DXOS.org
+//
+
+import React, { useCallback } from 'react';
+import { useDropzone } from 'react-dropzone';
+
+import { findAnnotation, type S } from '@dxos/effect';
+import { useTranslation } from '@dxos/react-ui';
+import { mx } from '@dxos/react-ui-theme';
+
+import { WNFS_PLUGIN } from '../meta';
+import { WnfsAction } from '../types';
+
+export type FileInputProps = {
+  schema: S.Schema.Any;
+  onChange: (file: File) => void;
+};
+
+export const FileInput = ({ schema, onChange }: FileInputProps) => {
+  const { t } = useTranslation(WNFS_PLUGIN);
+  const accept = findAnnotation<Record<string, string[]>>(schema.ast, WnfsAction.UploadAnnotationId);
+
+  const onDropAccepted = useCallback((files: File[]) => onChange?.(files[0]), [onChange]);
+
+  const { acceptedFiles, getRootProps, getInputProps, isFocused, isDragAccept, isDragReject } = useDropzone({
+    multiple: false,
+    accept,
+    onDropAccepted,
+  });
+
+  return (
+    <div
+      {...getRootProps()}
+      className={mx(
+        'flex flex-col items-center p-8 border border-separator',
+        isFocused && 'focus-ring',
+        isDragAccept && 'bg-attention',
```

Comment:
If this has text in it, also do `text-inverse`

Reply:
Wait, nevermind, I misread this as `bg-accent`, `bg-attention` is fine on its own of course.